tiledb/sm/array_schema/array_schema.h

```diff
@@ -102,19 +103,31 @@ class ArraySchema {
    * Returns a constant pointer to the selected attribute (nullptr if it
    * does not exist).
    */
-  const Attribute* attribute(unsigned int id) const;
+  shared_ptr<const Attribute> attribute(unsigned int id) const;
```

Comment:
Is there a good reason to expose `shared_ptr` in this way? It's only necessary if there are lifespan issues to deal with, and I don't see any.